shark_turbine/kernel/wave/codegen.py

```diff
@@ -830,17 +907,17 @@ def handle_reduction(emitter: WaveEmitter, node: fx.Node):
     # Without scheduling, we assume that we always start at 0.
     start = arith_d.constant(IndexType.get(), int(0))
 
-    idxc = IndexingContext.current()
-    tile_size = None
+    iterations = None
     for constraint in emitter.constraints:
         if isinstance(constraint, TilingConstraint) and constraint.dim == axis:
-            tile_size = constraint.tile_size.subs(idxc.subs)
-    assert tile_size is not None, "Could not find tiling constraint for reduction axis."
+            iterations = subs_idxc(constraint.iterations)
+    assert (
+        iterations is not None
+    ), "Could not find tiling constraint for reduction axis."
 
     # For now, we assume that dimensions that have tiling constraints on them,
     # do not have any other constraints.
-    dim = axis.subs(idxc.subs)
-    end = arith_d.constant(IndexType.get(), int(dim // tile_size))
+    end = arith_d.constant(IndexType.get(), int(iterations))
```

Comment:
Not for this PR, but wondering whether we should have subs_idxc return an int when we know the return type can be converted to an int? This way subs_idxc and safe_subs are guaranteed to give either floats/ints or symbolic expressions.